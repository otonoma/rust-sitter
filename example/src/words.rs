pub mod grammar {
    use rust_sitter::Rule;

    #[derive(Debug, Rule)]
    #[language]
    #[extras(
        re(r"\s")
    )]
    #[allow(dead_code)]
    pub struct Words {
        #[leaf("if")]
        keyword: (),
        #[word]
        #[leaf(pattern(r"[a-z_]+"))]
        word: String,
    }
}

#[cfg(test)]
mod tests {
    use super::*;
    use rust_sitter::Language;

    #[test]
    fn words_grammar() {
        insta::assert_debug_snapshot!(grammar::Words::parse("if"));
        insta::assert_debug_snapshot!(grammar::Words::parse("hello"));
        insta::assert_debug_snapshot!(grammar::Words::parse("ifhello"));
        insta::assert_debug_snapshot!(grammar::Words::parse("if hello"));
    }
}
