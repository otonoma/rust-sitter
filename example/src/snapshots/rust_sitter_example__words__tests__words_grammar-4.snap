---
source: example/src/words.rs
expression: "grammar::Words::parse(\"if hello\")"
---
ParseResult {
    result: Some(
        Words {
            keyword: (),
            word: "hello",
        },
    ),
    errors: [],
}
